sql/src/main/java/io/crate/execution/dml/upsert/GeneratedColumns.java

```diff
@@ -91,9 +91,10 @@ private void processGeneratedExpressionColumn(Reference reference,
         toInject.put(reference, expressionInput);
         final ColumnIdent ident = reference.column();
         if (!ident.isTopLevel()) {
-            List<Reference> references = nestedReferenceByTopLevelColumnName.getOrDefault(
-                ident.name(),
-                new ArrayList<>());
+            List<Reference> references = nestedReferenceByTopLevelColumnName.get(ident.name());
+            if (references == null) {
+                references = new ArrayList<>();
+            }
             references.add(reference);
             nestedReferenceByTopLevelColumnName.put(ident.name(), references);
```

Comment:
The put could also be moved up into the `== null` block, afterwards the reference that's already inside the dictionary is being mutated, so no additional `put` is necessary.